### PR TITLE
gactions: Add make test CI

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        mkdir -p /home/runner/go/src/github.com/kube-storage
+        cp -r /home/runner/work/volume-replication-operator/volume-replication-operator /home/runner/go/src/github.com/kube-storage
+
+    - name: run tests
+      working-directory: "/home/runner/go/src/github.com/kube-storage/volume-replication-operator"
+      env:
+        GOPATH: /home/runner/go
+      run: |
+        export PATH=$PATH:$GOPATH/bin
+        wget https://github.com/kubernetes-sigs/kubebuilder/releases/download/v2.3.2/kubebuilder_2.3.2_linux_amd64.tar.gz
```
Comment (Madhu-1) [inline .github/workflows/test.yaml:35]: read the version from a variable? or else if we need to change the version we need to change it in multiple places? we can do this enhancement later also.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  pull_request:
+    branches: [ '*' ]
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
```
Comment (Madhu-1) [inline .github/workflows/test.yaml:13]: running a test with 1.15 is fine?

**Review by Madhu-1 — APPROVED** [review]

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    - name: run tests
+      working-directory: "/home/runner/go/src/github.com/kube-storage/volume-replication-operator"
+      env:
+        GOPATH: /home/runner/go
+      run: |
+        export PATH=$PATH:$GOPATH/bin
+        export VERSION="2.3.2"
+        wget https://github.com/kubernetes-sigs/kubebuilder/releases/download/v"$VERSION"/kubebuilder_"$VERSION"_linux_amd64.tar.gz
+        tar -zxvf kubebuilder_"$VERSION"_linux_amd64.tar.gz
```
Comment (Madhu-1) [inline .github/workflows/test.yaml:37]: Does the version need to be separated by  `_` not `.`?

Comment (iamniting) [reply .github/workflows/test.yaml:37]: they have naming like this only.

**Review by Madhu-1 — APPROVED** [review]

**Review by sp98 — COMMENTED** [review]
nit
Suggestion to rename the file to unit-test.yaml

**Pull request merged**